# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1683901570 -7200
# Parent  0e4cbebb01795b215c52082339ea24aef35a5fa0
Feature: When multiple message are selected, current message is shown.

diff --git a/mail/app/profile/all-thunderbird.js b/mail/app/profile/all-thunderbird.js
--- a/mail/app/profile/all-thunderbird.js
+++ b/mail/app/profile/all-thunderbird.js
@@ -1407,16 +1407,19 @@ pref("app.donation.eoy.version.viewed", 
 // Use our own prefs for the end-of-year appeal. Otherwise running TB on the
 // profile would show the TB appeal and then never again the BB appeal.
 pref("betterbird.donation.eoy.version", 2022);
 pref("betterbird.donation.eoy.version.viewed", 0);
 pref("betterbird.donation.eoy.url", "https://www.betterbird.eu/end-of-year/");
 
 // Make this visible in the advanced preferences.
 pref("ui.prefersReducedMotion", 0);
+
+pref("mail.multiselect_message_shows_current", false);
+
 // IMAP-JS disabled, Bug 1707547.
 pref("mailnews.imap.jsmodule", false);
 
 // Unified toolbar
 
 // 0: icons beside text
 // 1: icons above text
 // 2: icons only
diff --git a/mail/base/content/about3Pane.js b/mail/base/content/about3Pane.js
--- a/mail/base/content/about3Pane.js
+++ b/mail/base/content/about3Pane.js
@@ -3105,17 +3105,23 @@ var threadPane = {
         messagePane.clearMessage();
         messagePane.clearMessages();
         return;
       case 1:
         let uri = gDBView.getURIForViewIndex(threadTree.selectedIndex);
         messagePane.displayMessage(uri);
         return;
       default:
-        messagePane.displayMessages(gDBView.getSelectedMsgHdrs());
+        if (Services.prefs.getBoolPref("mail.multiselect_message_shows_current", false)) {
+          let uri = gDBView.getURIForViewIndex(threadTree.currentIndex);
+          messagePane.displayMessage(uri);
+          return;
+        } else {
+          messagePane.displayMessages(gDBView.getSelectedMsgHdrs());
+        }
     }
   },
 
   /**
    * Handle threadPane drag events.
    */
   _onDragStart(event) {
     let row = event.target.closest(`tr[is^="thread-"]`);
@@ -4179,17 +4185,22 @@ var messagePane = {
 
     let msgHdr = top.messenger.msgHdrFromURI(messageURI);
     let index = threadTree.view.findIndexOfMsgHdr(msgHdr, true);
     if (index == nsMsgViewIndex_None) {
       // Change to correct folder if needed.
       displayFolder(msgHdr.folder.URI);
       index = threadTree.view.findIndexOfMsgHdr(msgHdr, true);
     }
-    threadTree.selectedIndex = index;
+    // The following line was added here:
+    // https://hg.mozilla.org/comm-central/rev/d0300a237b15#l1.14 and later changed here:
+    // https://hg.mozilla.org/comm-central/rev/a1d0dfa2895f#l1.20.
+    // Altogether it seems wrong that displaying a message selects it.
+    // After all, this is called from threadPane._onSelect().
+    // threadTree.selectedIndex = index;
 
     messageBrowser.contentWindow.displayMessage(messageURI, gViewWrapper);
     messageBrowser.hidden = false;
   },
 
   /**
    * Ensure the multi-message browser is not displaying messages.
    */
